src/JsonApiDotNetCore.MongoDb/MongoDbRepository.cs

```diff
@@ -50,20 +48,12 @@ public class MongoDbRepository<TResource, TId>
             CancellationToken cancellationToken)
         {
             if (layer == null) throw new ArgumentNullException(nameof(layer));
+            
+            var queryExpressionValidator = new MongoDbQueryExpressionValidator();
```

Comment:
This method calls ApplyQueryLayer, which performs the same so can be removed here.